tools/upgradetool/src/main/java/org/openhab/core/tools/internal/Upgrader.java

```diff
@@ -42,11 +43,15 @@
  * The {@link Upgrader} contains the implementation of the upgrade methods
  *
  * @author Jan N. Klug - Initial contribution
+ * @author Florian Hotze - Add script profile upgrade
  */
 @NonNullByDefault
 public class Upgrader {
     public static final String ITEM_COPY_UNIT_TO_METADATA = "itemCopyUnitToMetadata";
     public static final String LINK_UPGRADE_JS_PROFILE = "linkUpgradeJsProfile";
+    public static final String LINK_UPGRADE_SCRIPT_PROFILE = "linkUpgradeScriptProfile";
+    private static final Set<String> SCRIPT_PROFILE_NAMES = Set.of("transform:NASHORNJS", "transform:JS",
```

Comment:
Wouldn't it make more sense to use `transform:.*` and match a regex? Otherwise script profiles with JRule, JavaRule or other languages would not be transformed.

Reply:
Hmm, this should work as well :+1:

Reply:
Done.